### ISSUE TITLE
FC JS is not compatible with spice-html5 0.2.x+

### ISSUE THREAD

**Issue by stanislavlevin** [fleet-commander/fc-admin#241]
As of 0.2 `spice-html5` switched to JS modules.
Current FC (0.15.1) fails with:
```
SyntaxError: export declarations may only appear at top level of a module enums.js:26
SyntaxError: export declarations may only appear at top level of a module atKeynames.js:187
SyntaxError: import declarations may only appear at top level of a module utils.js:21
SyntaxError: import declarations may only appear at top level of a module png.js:25
SyntaxError: import declarations may only appear at top level of a module lz.js:21
SyntaxError: export declarations may only appear at top level of a module quic.js:1341
SyntaxError: import declarations may only appear at top level of a module bitmap.js:27
SyntaxError: export declarations may only appear at top level of a module spicedataview.js:122
SyntaxError: import declarations may only appear at top level of a module spicetype.js:28
SyntaxError: import declarations may only appear at top level of a module spicemsg.js:28
SyntaxError: import declarations may only appear at top level of a module wire.js:28
SyntaxError: import declarations may only appear at top level of a module spiceconn.js:30
SyntaxError: import declarations may only appear at top level of a module display.js:21
SyntaxError: import declarations may only appear at top level of a module port.js:22
SyntaxError: import declarations may only appear at top level of a module main.js:21
SyntaxError: import declarations may only appear at top level of a module inputs.js:21
SyntaxError: export declarations may only appear at top level of a module webm.js:651
SyntaxError: import declarations may only appear at top level of a module playback.js:26
SyntaxError: import declarations may only appear at top level of a module simulatecursor.js:29
SyntaxError: import declarations may only appear at top level of a module cursor.js:21
SyntaxError: export declarations may only appear at top level of a module jsbn.js:591
SyntaxError: import declarations may only appear at top level of a module rsa.js:37
SyntaxError: export declarations may only appear at top level of a module prng4.js:81
SyntaxError: import declarations may only appear at top level of a module rng.js:36
SyntaxError: export declarations may only appear at top level of a module sha1.js:348
SyntaxError: import declarations may only appear at top level of a module ticket.js:21
SyntaxError: export declarations may only appear at top level of a module resize.js:87
SyntaxError: export declarations may only appear at top level of a module filexfer.js:90
ReferenceError: SpiceConn is not definedspice_debug_bypass.js:22:1

```

**Comment by olivergs** [comment]
That file was created some time ago to avoid the spice-html5 client to stop when an error was caught. 
I'm checking if it is needed now and know if it is safe to remove it.

**Comment by olivergs** [comment]
I've pushed a patch to remove the file as it is not neccesary anymore.

Fixed in d7fc773e41cc0bd177c2422987d2dcb902716231 and 855e9cafc015bf6d25c5a2bf72fef4c2348c0bfa

**Comment by stanislavlevin** [comment]
Sorry, this issue is about JS modules. The removal of that file doesn't help.
```
SyntaxError: export declarations may only appear at top level of a module enums.js:26
SyntaxError: export declarations may only appear at top level of a module atKeynames.js:187
SyntaxError: import declarations may only appear at top level of a module utils.js:21
SyntaxError: import declarations may only appear at top level of a module png.js:25
SyntaxError: import declarations may only appear at top level of a module lz.js:21
SyntaxError: export declarations may only appear at top level of a module quic.js:1341
SyntaxError: import declarations may only appear at top level of a module bitmap.js:27
SyntaxError: export declarations may only appear at top level of a module spicedataview.js:122
SyntaxError: import declarations may only appear at top level of a module spicetype.js:28
SyntaxError: import declarations may only appear at top level of a module spicemsg.js:28
SyntaxError: import declarations may only appear at top level of a module wire.js:28
SyntaxError: import declarations may only appear at top level of a module spiceconn.js:30
SyntaxError: import declarations may only appear at top level of a module display.js:21
SyntaxError: import declarations may only appear at top level of a module port.js:22
SyntaxError: import declarations may only appear at top level of a module main.js:21
SyntaxError: import declarations may only appear at top level of a module inputs.js:21
SyntaxError: export declarations may only appear at top level of a module webm.js:651
SyntaxError: import declarations may only appear at top level of a module playback.js:26
SyntaxError: import declarations may only appear at top level of a module simulatecursor.js:29
SyntaxError: import declarations may only appear at top level of a module cursor.js:21
SyntaxError: export declarations may only appear at top level of a module jsbn.js:591
SyntaxError: import declarations may only appear at top level of a module rsa.js:37
SyntaxError: export declarations may only appear at top level of a module prng4.js:81
SyntaxError: import declarations may only appear at top level of a module rng.js:36
SyntaxError: export declarations may only appear at top level of a module sha1.js:348
SyntaxError: import declarations may only appear at top level of a module ticket.js:21
SyntaxError: export declarations may only appear at top level of a module resize.js:87
SyntaxError: export declarations may only appear at top level of a module filexfer.js:90
...

FC: Fatal error:ReferenceError: SpiceMainConn is not defined base.js:76:13
    logError https://fc.ipa.test:9090/cockpit/$a43681a7094931bf7ebed895245a44d79c5db2c2a6384c6ed80135f53ecf5c57/fleet-commander-admin/js/base.js:76
    FleetCommanderSpiceClient https://fc.ipa.test:9090/cockpit/$a43681a7094931bf7ebed895245a44d79c5db2c2a6384c6ed80135f53ecf5c57/fleet-commander-admin/js/fcspiceclient.js:131
    startSpiceHtml5 https://fc.ipa.test:9090/cockpit/$a43681a7094931bf7ebed895245a44d79c5db2c2a6384c6ed80135f53ecf5c57/fleet-commander-admin/js/livesession.js:107
    startLiveSession https://fc.ipa.test:9090/cockpit/$a43681a7094931bf7ebed895245a44d79c5db2c2a6384c6ed80135f53ecf5c57/fleet-commander-admin/js/livesession.js:265
    SessionStart https://fc.ipa.test:9090/cockpit/$a43681a7094931bf7ebed895245a44d79c5db2c2a6384c6ed80135f53ecf5c57/fleet-commander-admin/js/fcdbusclient.js:186
    s https://fc.ipa.test:9090/cockpit/$a43681a7094931bf7ebed895245a44d79c5db2c2a6384c6ed80135f53ecf5c57/base1/cockpit.js:979
    f https://fc.ipa.test:9090/cockpit/$a43681a7094931bf7ebed895245a44d79c5db2c2a6384c6ed80135f53ecf5c57/base1/cockpit.js:991
    n https://fc.ipa.test:9090/cockpit/$a43681a7094931bf7ebed895245a44d79c5db2c2a6384c6ed80135f53ecf5c57/base1/cockpit.js:897
```

This means that spice stuff should be `imported` and JS `scripts` marked as modules.

I could help later.

**Comment by stanislavlevin** [comment]
I would like to reopen this issue with a normal priority.

**Comment by olivergs** [comment]
I misunderstood the problem. I removed the script because the main error was on it. After that, I tested the spice client and all was working. Also I did not received that errors in console.

Is there a way to reproduce the problem so I can debug it?

**Comment by olivergs** [comment]
I'm currently working on this. I was updating the SPICE version wrongly until now. :facepalm: 

Now I've updated it but it will require some changes I'm working on now.

**Comment by olivergs** [comment]
Fixed in #247
